### ISSUE TITLE
"Display details" rounds to first decimal point

### ISSUE THREAD

**Issue by patrickhlauke** [ThePacielloGroup/CCA-Win#1]
In "Result - Luminosity" the "Contrast ratio" is shown with two decimal points. In the "Display details" this is rounded to only one decimal point. I'd like to see the latter match the former, particularly as for borderline fails (e.g. 4.46:1) the copy/paste-able text doesn't make sense (stating it's 4.5:1, then saying that's a fail)

![capture](https://cloud.githubusercontent.com/assets/895831/5297372/32860a8a-7ba6-11e4-8579-2b94318f8b68.PNG)

**Comment by matatk** [comment]
Apologies, I forgot this was a public repository and had already emailed @jun325 to request that the results should be rounded to 1 decimal place, [as this is what the W3C appears to use as the standard](http://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-contrast).

**Comment by ferllings** [comment]
The problem is what rounding method to use

**Comment by matatk** [comment]
I imagine the best way to do it is this:
- Don't do any rounding internally when doing the calculations.
- Round the result of the calculations to 1 decimal place for display purposes.
  - Using the [round half-way up method](http://en.wikipedia.org/wiki/Rounding#Round_half_up) I suppose.

I imagine this is how the W3C does it, from reading the docs, but if you know otherwise, or know anyone who may know definitively, please let us know :-).

**Comment by matatk** [comment]
@patrickhlauke, @ferllings, @stevefaulkner does the above suggestion with respect to rounding sound OK?  Having just asked @jun325 to work on it, it'd be good for us to be sure before he spends time on it :-).

**Comment by patrickhlauke** [comment]
looking at the algorithm like http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140916/G18 i don't get the feeling that WCAG has any kind of official stance on how many decimal points to use (and having a +0.05 in there suggests that at least sometimes you'd get at least two decimals). it may just be editiorial decision to then write 4.5:1 rather than 4.50:1
i'm easy either way, don't think in effect double digit precision are needed (and it would make me feel less bad about failing something that's 4.49:1 which effectively makes no difference in real money from 4.5:1).

I think @ferllings question was more about "should it round up, down, or what?". i'd say the rounding that is currently implemented for the text details ("round to 0 if < 5, round to 1 if >= 5") makes most sense.

The tool obviously needs to do the rounding BEFORE it makes a pass/fail determination (so we don't end up with 4.45-4.49 being first flagged as FAIL, and THEN rounded to 4.5)

**Comment by ferllings** [comment]
@patrickhlauke yes, I came across this issue with the CCA for OSX, and i haven't found any official statement for the number of decimal point nor the method to use for that (up, down, half even?)

I think your implementation suggestion makes sense.

**Comment by matatk** [comment]
The rounding method I linked to above is the same as that which Patrick suggests I think, so we're all agreed then; nice. Thanks for the quick responses.

**Comment by stevefaulkner** [comment]
is this done?

**Comment by matatk** [comment]
@stevefaulkner I believe @jun325 has updated it; I am currently in the process of integrating the new CCA with WAT.
